bubuku/features/swap_partitions.py

```diff
@@ -0,0 +1,180 @@
+from collections import namedtuple
+import logging
+from operator import attrgetter
+
+from bubuku.broker import BrokerManager
+from bubuku.controller import Check
+from bubuku.features.rebalance import BaseRebalanceChange
+from bubuku.zookeeper import BukuExhibitor
+
+_LOG = logging.getLogger('bubuku.features.swap_partitions')
+
+TpData = namedtuple('TpData', ('topic', 'partition', 'size', 'replicas'))
+
+
+class SwapPartitionsChange(BaseRebalanceChange):
+    def __init__(self, zk: BukuExhibitor, fat_broker_id: int, slim_broker_id: int, gap: int, size_stats: dict):
+        self.zk = zk
+        self.fat_broker_id = fat_broker_id
+        self.slim_broker_id = slim_broker_id
+        self.gap = gap
+        self.size_stats = size_stats
+        self.to_move = None
+
+    def run(self, current_actions):
+        if self.should_be_cancelled(current_actions):
+            _LOG.info("Cancelling swap partitions change as there are conflicting actions: {}".format(current_actions))
+            return False
+        try:
+            _LOG.info("Running swap partitions change: {}".format(self))
+            return self.__run_internal(current_actions)
+        except Exception:
+            _LOG.warn("Error occurred when performing partitions swap change", exc_info=True)
+            return False
+
+    def __run_internal(self, current_actions):
+        # if there is already a swap which was postponed - just execute it
+        if self.to_move:
+            return not self.__perform_swap(self.to_move)
+
+        # if there's a rebalance currently running - postpone current change
+        if self.zk.is_rebalancing():
+            return True
+
+        # merge topics size stats to a single dict
+        topics_stats = {}
+        for _, broker_stats in self.size_stats.items():
+            for topic in broker_stats["topics"].keys():
+                if topic not in topics_stats:
+                    topics_stats[topic] = {}
+                topics_stats[topic].update(broker_stats["topics"][topic])
+
+        # find partitions that are candidates to be swapped between "fat" and "slim" brokers
+        swap_partition_candidates = self.__find_all_swap_candidates(self.fat_broker_id, self.slim_broker_id,
+                                                                    topics_stats)
+
+        # smallest partition from slim broker is the one we move to fat broker
+        slim_broker_smallest_partition = min(swap_partition_candidates[self.slim_broker_id], key=attrgetter("size"))
+        if not slim_broker_smallest_partition:
+            _LOG.info("No partitions on slim broker(id: {}) found to swap".format(self.slim_broker_id))
+            return False
```

Comment:
Well... This makes no sense...
Suppose that I have newly deployed instance, It have no partitions at all. And it will never receive partitions. Is it fine?